pinot-common/src/main/java/com/linkedin/pinot/common/metadata/segment/SegmentZKMetadata.java

```diff
@@ -63,6 +69,16 @@ public SegmentZKMetadata(ZNRecord znRecord) {
     _totalRawDocs = znRecord.getLongField(CommonConstants.Segment.TOTAL_DOCS, -1);
     _crc = znRecord.getLongField(CommonConstants.Segment.CRC, -1);
     _creationTime = znRecord.getLongField(CommonConstants.Segment.CREATION_TIME, -1);
+
+    try {
+      String partitionMetadataJson = znRecord.getSimpleField(CommonConstants.Segment.PARTITION_METADATA);
+      if (partitionMetadataJson != null) {
+        _partitionMetadata = SegmentPartitionMetadata.fromJsonString(partitionMetadataJson);
+      }
+    } catch (IOException e) {
+      LOGGER.error("Exception caught while reading partition info from segment zk metadata.", e);
```

Comment:
Log the table name, or segment zk path name where the error happened. 

Consider setting partition data to null? I am sure we have code that brings all segment metadata in a loop (e.g. retention manager). Consider whether we want this one exception to stop processing other segments for this table (or worse, for all tables, giving up altogether, depending on how the caller is coded).